builder/optimizations.cc

```diff
@@ -489,15 +482,46 @@ class scope_reducer : public node_mutator {
     }
   }
 
-  void visit(const let_stmt* op) override { visit_stmt(op); }
-  void visit(const allocate* op) override { visit_stmt(op); }
-  void visit(const make_buffer* op) override { visit_stmt(op); }
-  void visit(const clone_buffer* op) override { visit_stmt(op); }
-  void visit(const crop_buffer* op) override { visit_stmt(op); }
-  void visit(const crop_dim* op) override { visit_stmt(op); }
-  void visit(const slice_buffer* op) override { visit_stmt(op); }
-  void visit(const slice_dim* op) override { visit_stmt(op); }
-  void visit(const truncate_rank* op) override { visit_stmt(op); }
+  void visit(const let_stmt* op) override {
+    std::vector<symbol_id> vars;
+    vars.reserve(op->lets.size());
+    for (const auto& s : op->lets) {
+      vars.push_back(s.first);
+    }
+    visit_stmt(op, vars);
+  }
+  void visit(const allocate* op) override {
+    symbol_id vars[] = {op->sym};
+    visit_stmt(op, vars);
```

Comment:
I think an overload with the old interface would be worth it to avoid this ~4x increase in code noise for these visitors (except `let_stmt` of course).